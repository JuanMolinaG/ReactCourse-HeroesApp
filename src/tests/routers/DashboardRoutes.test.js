import React from 'react';
import { mount } from 'enzyme';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';

describe('Prueba en <DashboardRoutes />', () => {
  test('Debe de mostrar el componente correctamente', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: 'Juan G.',
        logged: true,
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Juan G.');
  });
});
