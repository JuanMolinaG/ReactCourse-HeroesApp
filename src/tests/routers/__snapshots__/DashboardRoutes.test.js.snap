// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <DashboardRoutes /> Debe de mostrar el componente correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Juan G.
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    Array [
      <h1>
        Marvel Screen
      </h1>,
      <hr />,
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Daredevil #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
